src/views/Dog/Dog.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { fetchDogById } from '../../services/dog';
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
+import DogDetails from '../../components/DogDetails/DogDetails';
+import './Dog.css';
+
+export default function Dog() {
+  const [dog, setDog] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const params = useParams();
```

Comment:
we can de-structure id here as we use it in multiple places 
`const { id } = useParams()`